app/models/robot_world.rb

```diff
@@ -1,56 +1,80 @@
+require_relative 'statbot'
 class RobotWorld
-  attr_reader :world
-  def initialize(world)
-    @world = world
+  include StatBot
+  attr_reader :database, :robot_ages, 
+              :max_hires, :hires_by_year,
+              :by_department, :by_city, :by_state
+  def initialize(database)
+    @database = database
+    @robot_ages = {"years" => [], "months" => [], "days" => []}
+    @hires_by_year = {}
   end
 
   def create(robot)
-    world.transaction do
-      world['robots'] ||= []
-      world['population'] ||= 0
-      world['population'] += 1
-      world['robots'] << new_robot(world['population'], robot[:name], robot[:city], robot[:state],
-                                   robot[:birthdate], robot[:date_hired], robot[:department], "https://robohash.org/#{world['population']}")
-    end
+    robo = {:name => robot["name"], :city => robot["city"], 
```

Comment:
is it necessary to create this hash here? why not just access the specific pieces of data within your SQL query below?